src/equations/compressible_euler_multicomponent_2d.jl

```diff
@@ -0,0 +1,713 @@
+
+@doc raw"""
+    CompressibleEulerMulticomponentEquations2D
+
+!!! warning "Experimental code"
+    This system of equations is experimental and can change any time.
+
+Multicomponent version of the compressible Euler equations for an ideal gas in two space dimensions.
+"""
+struct CompressibleEulerMulticomponentEquations2D{RealT<:Real} <: AbstractCompressibleEulerMulticomponentEquations{2, 5}
+  gamma1          ::RealT 
+  gamma2          ::RealT
+  gas_constant1   ::RealT
+  gas_constant2   ::RealT
+  cv1             ::RealT
+  cv2             ::RealT
+  cp1             ::RealT
+  cp2             ::RealT
+end
+
+function CompressibleEulerMulticomponentEquations2D()
+
+  # Set ratio of specific heat of the gas per species 
+  # (It holds: gamma = cp/cv) (Can be used for consistency check!)
+  gamma1  = 1.4
+  gamma2  = 1.4
+  
+  # Set specific gas constant with respect to the molar mass per species 
+  # (It holds: gas_constant = R/M, with R = molar gas constant, M = molar mass) 
+  # (For a calorically and thermally perfect gas Mayer's relation holds: gas_constant = cp - cv) (Can be used for consistency check!)
+  gas_constant1     = 0.4
+  gas_constant2     = 0.4
+
+  # Set specific heat for a constant volume per species
+  cv1    = 1.0
+  cv2    = 1.0
+
+  # Set specific heat for a constant pressure per species
+  cp1    = 1.4
+  cp2    = 1.4
+
+  CompressibleEulerMulticomponentEquations2D(gamma1, gamma2, gas_constant1, gas_constant2, cv1, cv2, cp1, cp2)
+end
+
+
+function CompressibleEulerMulticomponentEquations2D(gamma1, gamma2, gas_constant1, gas_constant2)
+  
+  # Set specific heat for a constant volume per species
+  cv1    = gas_constant1 / (gamma1 - 1.0)
+  cv2    = gas_constant2 / (gamma2 - 1.0)
+
+  # Set specific heat for a constant pressure per species
+  cp1    = gas_constant1 + cv1 
+  cp2    = gas_constant2 + cv2
+
+  CompressibleEulerMulticomponentEquations2D(gamma1, gamma2, gas_constant1, gas_constant2, cv1, cv2, cp1, cp2)
+end
+
+
+function CompressibleEulerMulticomponentEquations2D(gamma1, gamma2, gas_constant1, gas_constant2, cv1, cv2, cp1, cp2)
+
+  if !isapprox(gamma1, cp1/cv1) 
+    error("Your values for gamma1, cp1, cv1 are not consistent!")
+  end 
+
+  if !isapprox(gamma2, cp2/cv2)  
+    error("Your values for gamma2, cp2, cv2 are not consistent!")
+  end 
+
+  if !isapprox(gas_constant1, cp1 - cv1)
+    error("Your values for gas_contant1, cp1, cv1 are not consistent!")
+  end
+
+  if !isapprox(gas_constant2, cp2 - cv2)
+    error("Your values for gas_contant2, cp2, cv2 are not consistent!")
+  end
+
+  CompressibleEulerMulticomponentEquations2D(gamma1, gamma2, gas_constant1, gas_constant2, cv1, cv2, cp1, cp2) = new(gamma1, gamma2, gas_constant1, gas_constant2, cv1, cv2, cp1, cp2)
+end
```

Comment:
This won't (shouldn't) work! The method `new` is only available in inner constructors, however, for a constructor to be an **inner** constructor, it needs to be defined within the struct. Quoting from the [docs](https://docs.julialang.org/en/v1/manual/constructors/#man-inner-constructor-methods) (highlights are mine): 
> An inner constructor method is like an outer constructor method, except for two differences: [...] It is **declared inside the block of a type declaration**, rather than outside of it like normal methods.

Also, this constructor is never used in any test. You might thus want consider adding some [manual tests](https://github.com/trixi-framework/Trixi.jl/blob/multicomponentJoinsTaal/test/test_manual.jl) for it to also trigger the different error conditions.

Reply:
You're right. I deleted (see e3bb50f) this constructor for now, but it's on my list for the next pull request.